### PR TITLE
Allows to have SVG URL

### PR DESCRIPTION
**Pull request by sbrunner**
Test URL: https://camptocamp.github.io/ngeo/svg-url/examples/contribs/gmf/apps/desktop_alt.html?map_crosshair=true
https://camptocamp.github.io/ngeo/svg-url/examples/svg.html

### CODE REVIEW

File: contribs/gmf/apps/desktop_alt/Controller.js [hunk 1]
```diff
@@ -33,6 +37,16 @@ if (!window.requestAnimationFrame) {
   window.location.href = 'https://geomapfish.org/';
 }
 
+module.value('gmfPermalinkOptions', /** @type {import('gmf/permalink/Permalink.js').PermalinkOptions} */ ({
```
Comment (pfirpfel) [inline contribs/gmf/apps/desktop_alt/Controller.js]: Move further below where `module` is declared. Can't compile otherwise.

Comment (sbrunner) [reply contribs/gmf/apps/desktop_alt/Controller.js]: Fixed :-)

File: contribs/gmf/apps/desktop_alt/image/logo.svg [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+         style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none"
+         id="path32"
+         inkscape:connector-curvature="0" /><path
+         d="m 184.6755,282.8835 v -2.832 h -34.656 v 2.832 h 12.048 c 0,0 -1.872,4.032 -1.872,7.776 0,0 0,4.8 2.544,6.408 0,0 2.544,1.608 9.312,1.608 h 12.624 v -2.88 h -12.48 c 0,0 -5.52,0 -7.464,-1.032 0,0 -1.944,-1.032 -1.944,-4.488 0,0 0,-1.68 0.408,-3.528 0,0 0.408,-1.848 0.84,-2.856 l 0.384,-1.008 h 20.256"
+         style="fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none"
+         id="path34"
+         inkscape:connector-curvature="0" /></g></g></svg>
```
Comment (pfirpfel) [inline contribs/gmf/apps/desktop_alt/image/logo.svg]: Needs a new line at the end, otherwise the linter fails.

**Comment by sbrunner** [comment]
Yes It's also possible to use the python URL resolver with the following config:
```interfaces_config.<interface>.static: <package>_geoportal:static/images/img.svg```

For me booth should work...

**Comment by sbrunner** [comment]
I just fixed the issue :-)

**Comment by pfirpfel** [comment]
Using this branch, I used `make serve-gmf-apps` and opened this url:
https://localhost:3000/contribs/gmf/apps/desktop_alt.html?map_x=643288&map_y=193988&map_crosshair=true

No crosshair for me.

The browser requests this:

`https://localhost:3000/contribs/gmf/apps/%3Csvg%20viewBox=%220%200%2022%2022%22%20xmlns=%22http://www.w3.org/2000/svg%22%3E%3Cpath%20d=%22M10.963.4A2.5%202.5%200%200%200%208.5%202.935v5.55H2.95a2.5%202.5%200%201%200%200%205H8.5v5.554a2.5%202.5%200%201%200%205%200v-5.553h5.55a2.5%202.5%200%201%200%200-5H13.5v-5.55A2.5%202.5%200%200%200%2010.963.4z%22%20color=%22`

This is `https://localhost:3000/contribs/gmf/apps/`plus the beginning of the svg.

**Comment by sbrunner** [comment]
Do you use the latest version?
With the dev server it's normal that you don't have the icon, but you should have a 404 error on the svg file...

**Comment by pfirpfel** [comment]
Yes, I just tried again. Now it is just a 404

**Comment by sbrunner** [comment]
:-)

**Comment by sbrunner** [comment]
Fill also fix: https://github.com/camptocamp/ngeo/pull/4940

File: buildtools/svg-viewbox-loader.js [hunk 3]
```diff
@@ -1,32 +1,79 @@
...
-        })
-        tag.attributes.push(['viewBox', `0 0 ${width} ${height}`, true]);
-        tag.attributes.push(['height', '1em', true]);
-        tag.attributes.push(['width', `${width / height}em`, true]);
+          return attribute[0] !== 'width' && attribute[0] != 'height' && attribute[0] != 'viewBox';
+        });
+        if (x) {
+          tag.attributes.push(['x', x, true]);
+        }
+        if (y) {
```
Comment (gberaudo) [inline buildtools/svg-viewbox-loader.js]: x or y could be 0, right?

File: contribs/gmf/apps/desktop_alt/Controller.js [hunk 4]
```diff
@@ -20,7 +20,11 @@ import ngeoRoutingModule from 'ngeo/routing/module.js';
 import EPSG2056 from '@geoblocks/proj/src/EPSG_2056.js';
 import EPSG21781 from '@geoblocks/proj/src/EPSG_21781.js';
 import ngeoStatemanagerWfsPermalink from 'ngeo/statemanager/WfsPermalink.js';
-import {Circle, Fill, Stroke, Style} from 'ol/style';
+import Style from 'ol/style/Style.js';
```
Comment (gberaudo) [inline contribs/gmf/apps/desktop_alt/Controller.js]: Why change this?

File: examples/svg.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+<html ng-app='app'>
+  <head>
+    <title>SVG example</title>
+    <meta charset="utf-8">
+    <meta name="viewport"
+          content="initial-scale=1.0, user-scalable=no, width=device-width">
+    <meta name="mobile-web-app-capable" content="yes">
+  </head>
+  <body ng-controller="MainController as ctrl">
+    <p id="desc">This example shows deferent way to include an SVG in the application using Webpack.</p>
```
Comment (gberaudo) [inline examples/svg.html]:
```suggestion
    <p id="desc">This example shows different ways to include an SVG in the application using Webpack.</p>
```

File: buildtools/generate.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
+// Initially get from https://github.com/tildeio/simple-html-tokenizer/blob/v0.1.1/lib/simple-html-tokenizer/generator.js
```
Comment (gberaudo) [inline buildtools/generate.js]: Could you give a clearer filename?

File: buildtools/svg-viewbox-loader.js (hunk 3 above)
Comment (sbrunner) [reply buildtools/svg-viewbox-loader.js]: Yes but as I understand it's the default value then we don't need to specify it ...

File: contribs/gmf/apps/desktop_alt/Controller.js (hunk 4 above)
Comment (sbrunner) [reply contribs/gmf/apps/desktop_alt/Controller.js]: To use the right files directly without redirect :-)

File: examples/svg.html (hunk 5 above)
Comment (sbrunner) [reply examples/svg.html]: Thanks :-)

File: buildtools/generate.js (hunk 6 above)
Comment (sbrunner) [reply buildtools/generate.js]: like `generate-html-grom-tockens.js` ?

Comment (gberaudo) [reply buildtools/generate.js]: generate-xml-from-tokens.js

**Pull request merged**